File: python/PetShop.py
# definisi kelas petshop
class PetShop:
    # inisialisasi atribut privat
    __id = -1
    __nama_produk = ""
    __kategori = ""
    __harga = 0

    # konstruktor untuk inisialisasi objek petshop
    def __init__(self, id=-1, nama_produk="", kategori="", harga=0):
        self.__id = id
        self.__nama_produk = nama_produk
        self.__kategori = kategori
        self.__harga = harga

    # metode untuk mendapatkan id produk
    def get_id(self):
        return self.__id

    # metode untuk mengatur id produk
    def set_id(self, id):
        self.__id = id

    # metode untuk mendapatkan nama produk
    def get_nama_produk(self):
        return self.__nama_produk

    # metode untuk mengatur nama produk
    def set_nama_produk(self, nama_produk):
        self.__nama_produk = nama_produk

    # metode untuk mendapatkan kategori produk
    def get_kategori(self):
        return self.__kategori

    # metode untuk mengatur kategori produk
    def set_kategori(self, kategori):
        self.__kategori = kategori

    # metode untuk mendapatkan harga produk
    def get_harga(self):
        return self.__harga

    # metode untuk mengatur harga produk
    def set_harga(self, harga):
        self.__harga = harga

File: java/PetShop.java
class PetShop {
    private String id; // variabel untuk menyimpan ID produk
    private String namaProduk; // variabel untuk menyimpan nama produk
    private String kategoriProduk; // variabel untuk menyimpan kategori produk
    private double hargaProduk; // variabel untuk menyimpan harga produk

    // konstruktor untuk inisialisasi objek PetShop
    public PetShop(String id, String namaProduk, String kategoriProduk, double hargaProduk) {
        this.id = id; // menginisialisasi ID produk
        this.namaProduk = namaProduk; // menginisialisasi nama produk
        this.kategoriProduk = kategoriProduk; // menginisialisasi kategori produk
        this.hargaProduk = hargaProduk; // menginisialisasi harga produk
    }

    // getter untuk mendapatkan ID produk
    public String getId() {
        return id;
    }

    // setter untuk mengubah ID produk
    public void setId(String id) {
        this.id = id;
    }

    // getter untuk mendapatkan nama produk
    public String getNamaProduk() {
        return namaProduk;
    }

    // setter untuk mengubah nama produk
    public void setNamaProduk(String namaProduk) {
        this.namaProduk = namaProduk;
    }

    // getter untuk mendapatkan kategori produk
    public String getKategoriProduk() {
        return kategoriProduk;
    }

    // setter untuk mengubah kategori produk
    public void setKategoriProduk(String kategoriProduk) {
        this.kategoriProduk = kategoriProduk;
    }

    // getter untuk mendapatkan harga produk
    public double getHargaProduk() {
        return hargaProduk;
    }

    // setter untuk mengubah harga produk
    public void setHargaProduk(double hargaProduk) {
        this.hargaProduk = hargaProduk;
    }
}